iml-api/src/timer.rs

```diff
@@ -0,0 +1,105 @@
+use crate::error::ImlApiError;
+use iml_manager_client::{delete, get_client, put};
+use iml_manager_env::{get_timer_addr, running_in_docker};
+use std::time::Duration;
+
+#[derive(serde::Serialize, Debug)]
+pub struct TimerConfig {
+    config_id: String,
+    file_prefix: String,
+    timer_config: String,
+    service_config: String,
+}
+
+pub async fn configure_snapshot_timer(
+    config_id: i32,
+    fsname: String,
+    interval: Duration,
+    use_barrier: bool,
+) -> Result<(), ImlApiError> {
+    let iml_cmd = format!(
+        r#"/bin/bash -c "/usr/bin/date +\"%%Y-%%m-%%dT%%T%%:z\" | xargs -I % echo \"iml snapshot create {} -c 'automatically created by IML' {} {}-{}-%\"""#,
```

Comment:
Isn't it better to write and use custom daemon instead of messing with systemd timers?

Reply:
Each has its strengths but the systemd timer makes sense here in my opinion. We are using it for stratagem as well.